acceptance/put_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+// +build acceptance
+
+package acceptance
+
+import (
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/acceptance/localcluster"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// TestPut starts up an N node cluster and runs N workers that write
+// to independent keys.
+func TestPut(t *testing.T) {
+	l := localcluster.Create(*numNodes, stopper)
+	l.Start()
+	defer l.Stop()
+
+	checkRangeReplication(t, l, 20*time.Second)
+
+	c, err := makeDBClient(l, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	r, _ := util.NewPseudoRand()
+	value := util.RandBytes(r, 8192)
+
+	errs := make(chan error, *numNodes)
+	start := time.Now()
+	deadline := start.Add(*duration)
+	var count int64
+	for i := 0; i < *numNodes; i++ {
+		go func(i int) {
+			for time.Now().Before(deadline) {
+				k := atomic.AddInt64(&count, 1)
+				v := value[:r.Intn(len(value))]
+				if _, err := c.Put(fmt.Sprintf("%08d", k), v); err != nil {
+					errs <- err
+					return
+				}
+			}
+			errs <- nil
+		}(i)
+	}
+
+	for i := 0; i < *numNodes; {
+		delay := 1 * time.Second
+		if i > 1 {
+			delay = 1 * time.Hour
```

Comment:
You're reading it wrong. :)

The `delay` is how long to wait between logging test progress. We log progress every second until the first worker exits at which point we set the delay really large in order to not log progress anymore. The code is wrong. It should really read `i > 0`. And in practice it doesn't seem to matter as the workers all exit nearly simultaneously. I think I'll just get rid of that.